pallets/nft/src/lib.rs

```diff
@@ -0,0 +1,297 @@
+        ///
+        /// # Permissions
+        /// * Asset
+        /// * Portfolio
+        #[weight = <T as Config>::WeightInfo::mint_nft(nft_metadata_attributes.len() as u32)]
+        pub fn mint_nft(origin, ticker: Ticker, nft_metadata_attributes: Vec<NFTMetadataAttribute>) -> DispatchResult {
+            Self::base_mint_nft(origin, ticker, nft_metadata_attributes)
+        }
+
+        /// Burns the given NFT from the caller's portfolio.
+        ///
+        /// # Arguments
+        /// * `origin` - is a signer that has permissions to act as an agent of `ticker`.
+        /// * `ticker` - the ticker of the NFT collection.
+        /// * `nft_id` - the id of the NFT to be burned.
+        /// * `portfolio_kind` - the portfolio that contains the nft.
+        ///
+        /// ## Errors
+        /// - `CollectionNotFound` - if the collection associated to the given ticker has not been created.
+        /// - `NFTNotFound` - if the given NFT does not exist in the portfolio.
+        ///
+        /// # Permissions
+        /// * Asset
+        /// * Portfolio
+        #[weight = <T as Config>::WeightInfo::burn_nft()]
+        pub fn burn_nft(origin, ticker: Ticker, nft_id: NFTId, portfolio_kind: PortfolioKind) -> DispatchResult {
+            Self::base_burn_nft(origin, ticker, nft_id, portfolio_kind)
+        }
+    }
+}
+
+decl_error! {
+    pub enum Error for Module<T: Config> {
+        /// An overflow while calculating the balance.
+        BalanceOverflow,
+        /// An underflow while calculating the balance.
+        BalanceUnderflow,
+        /// The ticker is already associated to an NFT collection.
+        CollectionAlredyRegistered,
+        /// The NFT collection does not exist.
+        CollectionNotFound,
+        /// A duplicate metadata key has been passed as parameter.
+        DuplicateMetadataKey,
+        /// The associated asset is not of type NFT.
+        InvalidAssetType,
+        /// Either the number of keys or the key identifier does not match the keys defined for the collection.
+        InvalidMetadataAttribute,
+        /// The maximum number of metadata keys was exceeded.
+        MaxNumberOfKeysExceeded,
+        /// The NFT does not exist.
+        NFTNotFound,
+        /// At least one of the metadata keys has not been registered.
+        UnregisteredMetadataKey,
+    }
+}
+
+impl<T: Config> Module<T> {
+    fn base_create_nft_collection(
+        origin: T::Origin,
+        ticker: Ticker,
+        collection_keys: NFTCollectionKeys,
+    ) -> DispatchResult {
+        // Verifies if the caller has the right permissions to create the collection
+        let _did = Identity::<T>::ensure_perms(origin.clone())?;
+
+        // Verifies if the ticker is already associated to an NFT collection
+        if CollectionTicker::contains_key(&ticker) {
+            return Err(Error::<T>::CollectionAlredyRegistered.into());
+        }
```

Comment:
Please use the `ensure` macro for these checks.  This is the code-style commonly used in Substrate pallets.
```suggestion
        ensure!(!CollectionTicker::contains_key(&ticker), Error::<T>::CollectionAlredyRegistered);
```